suite-common/connect-init/src/connectInitThunks.ts

```diff
@@ -18,157 +20,163 @@ const CONNECT_INIT_MODULE = '@common/connect-init';
 // If you are looking where connectInitSettings is defined, it is defined in packages/suite/src/support/extraDependencies.ts
 // or in suite-native/state/src/extraDependencies.ts depends on which platform this connectInitThunk runs.
 
-export const connectInitThunk = createThunk(
-    `${CONNECT_INIT_MODULE}/initThunk`,
-    async (_, { dispatch, getState, extra }) => {
-        const {
-            selectors: {
-                selectEnabledNetworks,
-                selectIsPendingTransportEvent,
-                selectDebugSettings,
-            },
-            actions: { lockDevice },
-            utils: { connectInitSettings },
-        } = extra;
-
-        const getEnabledNetworks = () => selectEnabledNetworks(getState());
-
-        // set event listeners and dispatch as
-        TrezorConnect.on(DEVICE_EVENT, ({ event: _, ...eventData }) => {
-            if (eventData.type === DEVICE.CONNECT || eventData.type === DEVICE.CONNECT_UNACQUIRED) {
-                // This special case here allows us to "inject" extra data into action's payload
-                // and change the type of the action (in this case DeviceEvent type !== Redux Action type)
-                dispatch(deviceConnectThunks({ type: eventData.type, device: eventData.payload }));
-            } else {
-                // dispatch event as action
-                dispatch({ type: eventData.type, payload: eventData.payload });
-            }
-        });
-
-        TrezorConnect.on(UI_EVENT, ({ event: _, ...action }) => {
-            if (action.type === 'ui-select_device') {
-                // this is why you received the ui-select_device event.
-                console.warn(
-                    'Hey, it looks like you called a TrezorConnect method without providing device property.',
-                );
-            }
-            // dispatch event as action
-            dispatch(action);
-        });
-
-        TrezorConnect.on(TRANSPORT_EVENT, ({ event: _, ...action }) => {
-            // dispatch event as action
-            dispatch(action);
-        });
-
-        TrezorConnect.on(BLOCKCHAIN_EVENT, ({ event: _, ...action }) => {
+export const connectInitThunk = createThunk<
+    void,
+    | {
```

Comment:
Unfortunatell,y I didn't see a better way how to "hook" into this process here. I really, really wanted to avoid using middleware. IMHO this is better than making the middleware hell worse.

In the ideal world, this would be modular.